pkg/reader/usdc_reader.go

```diff
@@ -111,8 +112,9 @@ func NewUSDCMessageReader(
 				lggr:           lggr,
 				contractReader: contractReaders[chainSelector],
 				cctpDestDomain: domains,
-				boundContract:  contract,
+				boundContract:  contract, // TODO: this is not needed if we switch to the Extended contract reader.
```

Comment:
This is one reason we might want to use `Extended`. The other is to ensure fault violation detection is enabled.